### PR TITLE
fix(inappbrowser): handle redirect links and improve download/notification flow (Android)

### PR DESCRIPTION
**Pull request by sgomdie**
Fixes handling of 3xx redirects in InAppBrowser on Android and routes download-triggering redirects to a DownloadHandler that uses DownloadManager with fallbacks. Adds JS events: `downloadstart`, `downloadprogress`, `downloadcomplete`, `downloaderror`.

Files to review: JS wrapper (`www/inappbrowser.js` or `src/index.ts`), Android sources (InAppBrowserClient / DownloadHandler), `AndroidManifest.xml`.

iOS implementation is NOT included in this PR and will be submitted in a follow-up PR.

## Quick test steps
1. npm ci && npx cap sync; run Android (emulator/device API >= 26).  
2. Open a URL that returns a 302/3xx → verify final URL shown and events fire (`onLoadStart/onLoadStop` or equivalent).  
3. Open a link with `Content-Disposition: attachment` → verify `downloadstart`, `downloadprogress`, `downloadcomplete` (file saved) and `downloaderror` on failure.

Edge cases to test:
- Multiple chained redirects (2+ 3xx hops).  
- Downloads started via redirect responses.  
- Permission-denied flows on Android (simulate and verify fallback and user message).

## CI / Permissions notes
- Review AndroidManifest changes. Prefer MediaStore/SAF on Android 10+ instead of WRITE_EXTERNAL_STORAGE if possible.  
- Ensure runtime permission requests are documented when required.  
- For Android CI include `npm ci`, `npx cap sync` and `./gradlew assembleDebug` in build steps.

## CHANGELOG entry (suggested)
- Fixed: Proper handling of redirect links and improved download + notification handling on Android (preserves headers, shows progress and errors). iOS implementation pending.

## Checklist
- [ ] No sensitive tokens/headers are logged or leaked.
- [ ] Manifest changes reviewed and approved.
- [ ] Manual Android validation completed.
- [ ] CHANGELOG/README updated.
- [ ] iOS work tracked for follow-up PR.

## Summary by CodeRabbit

* **New Features**
  * Native download support with automatic fallback paths
  * File picker enhancements with improved MIME detection and prompts
  * Download completion notifications allowing open/share actions

* **Improvements**
  * Better error handling and lifecycle management to reduce crashes
  * Enabled cookies and third-party cookies for improved web compatibility
  * More reliable filename and MIME-type detection

* **Chores**
  * Added a formatting script for code style fixes

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,12 +59,12 @@
     "eslint": "^8.57.0",
     "eslint-plugin-import": "^2.31.0",
     "husky": "^9.1.7",
-    "prettier": "^3.4.2",
```
Comment (riderx) [inline package.json]: Can you explain why this changes ?

File: package-lock.json
Comment (riderx) [inline package-lock.json:1]: we use bun not npm please remove this file :)

File: android/src/main/java/ee/forgr/capacitor_inappbrowser/WebViewDialog.java [hunk 2]
```diff
@@ -2994,6 +3372,820 @@ private File createImageFile() throws IOException {
         return image;
     }
 
+    private String downloadReasonToString(int reason) {
```
Comment (riderx) [inline android/src/main/java/ee/forgr/capacitor_inappbrowser/WebViewDialog.java:3375]: i think the whole download manager should have it's own file, also what about ios does it work properly already?

File: package.json (hunk 1 above)
Comment (sgomdie) [reply package.json]: Yeah because I saw that prettier versions are differentes from github CI that package.json and in local when i check the project with prettier have different warns. I changed to have the same versions in local than in github with bun. Think is neccesary.

File: package-lock.json
Comment (sgomdie) [reply package-lock.json:1]: Okay! Going to do that right now

Comment (sgomdie) [reply package-lock.json:1]: I saw that package.lock was already in git. Only changed versions related with prettier.

File: android/src/main/java/ee/forgr/capacitor_inappbrowser/WebViewDialog.java (hunk 2 above)
Comment (sgomdie) [reply android/src/main/java/ee/forgr/capacitor_inappbrowser/WebViewDialog.java:3375]: We can plan how to do it better okay. I have more changes prepared to fix this on ios. My teammate worked on it. We planned to push in a separate PR. Do you want to check both at the same PR?

**Comment by sgomdie** [comment]
Hello! Added some comments too. Lets have a talk and we can try to close this PR :)

File: android/src/main/java/ee/forgr/capacitor_inappbrowser/WebViewDialog.java (hunk 2 above)
Comment (riderx) [reply android/src/main/java/ee/forgr/capacitor_inappbrowser/WebViewDialog.java:3375]: let's make PR atomic please otherwise it will be a mess to review :)

File: package.json (hunk 1 above)
Comment (riderx) [reply package.json]: ok thanks

Comment (riderx) [reply package.json]: we should not have lock i removed it

**Comment by riderx** [comment]
Can you resolve conflict. ?
